### PR TITLE
Delete StockDistribution elements when quantity is set to zero.

### PR DESCRIPTION
**Pull request by fcapovilla**
Tentative pull request to fix Issue #135 

StockDistribution elements are now automatically deleted when their quantity gets set to zero.
This should remove them from the finalized order.

Please review these changes since I'm not 100% sure how the system is supposed to work.

### CODE REVIEW

**Comment by bglusman** [comment]
@fcapovilla thanks so much for the PR!  Happy Hacktoberfest!  I'm at work now but at a glance this looks great, thanks so much!  Will download and run locally later and/or perhaps @komizutama will take a look and make sure it works/looks OK behavior wise to them, but really excited to have our first Hactoberfest contribution, and hope you'll consider sticking around!

File: assets/js/facility.js
Comment (bglusman) [inline assets/js/facility.js]: May be cleaner to add this as fallback behavior in intFromFindClass since it would make behavior more consistent there

File: lib/open_pantry/food_selection.ex
Comment (bglusman) [inline lib/open_pantry/food_selection.ex:82]: Excellent, thank you for adding a Dialyzer spec!  I have another issue looking for help with getting these more pervasive and useful in the app, do you have any experience with Dialyzer?  Ideally I'd also like to learn more about how to understand it and do it myself, and perhaps make some PR's the Dialyxir to give more readable/useful error messages (I did something similar for ex_crypto when I found Erlang's error behavior in :crypto module confusing), but depending on your time and aptitude, anything is appreciated!

File: lib/open_pantry/web/views/shared_view.ex
Comment (bglusman) [inline lib/open_pantry/web/views/shared_view.ex:16]: Awesome... I haven't done much rescuing in Elixir, but this seems like a good use case for it :-)  I imagine you hit this by necessity after other parts of fix were done?  I'd like to improve our test coverage, I had some better integration tests before the CSS redesign, and had trouble getting them to work afterwards.  There's also an open PR to move wallaby to use headless chrome that might help in fixing up integration tests if you have any interest in helping to take over that, but I think though it mostly works locally there's some work to be done getting it to work on Travis or CircleCI (I think we run both at the moment, but Circle has some issues and Travis I got working faster)

**Review by bglusman — APPROVED** [review]
Wouldn't mind a small fix on the JS `|| 0` issue, but depending on your time I'm also happy to merge this in as is and possibly change later on myself, it doesn't block the PR and I ran it locally and works fine for me.  Thanks so much Frédérick!

File: assets/js/facility.js
Comment (fcapovilla) [reply assets/js/facility.js]: I agree.
I put the fallback there because I wanted to avoid modifying existing code as much as possible.
I'll move the fallback behavior in intFromFindClass.

File: lib/open_pantry/food_selection.ex
Comment (fcapovilla) [reply lib/open_pantry/food_selection.ex:82]: Sadly, I do not have experience with Dialyzer, but I also want to learn it.
I might look into the issue if I have time.

File: assets/js/facility.js
Comment (bglusman) [reply assets/js/facility.js]: Appreciate the thought, but it's a young app, plenty of opportunity to improve things, and if something breaks, well, we may learn something :-)  Obviously would be nice if test coverage gave more confidence, right now if only detects certain egregious breaks, hence a little manual playing before merging a PR, but a lots still in flux.  Also, if you know any local food banks/pantries in Quebec that might be interested and/or can help us translate into french, having a second language more usably translated may help figure out some of the patterns etc going forward, as it really needs to be very multi-language, but the current translation style from files probably won't work for that long term.

File: lib/open_pantry/web/views/shared_view.ex
Comment (fcapovilla) [reply lib/open_pantry/web/views/shared_view.ex:16]: Yes, added this by necessity because deleted StockDistributions could not be rendered for the channel's response.
In the background, "StockDistribution.find" uses "Repo.one!" which raises exceptions instead of returning tuples, so rescuing seemed like a good match.

**Comment by fcapovilla** [comment]
Added a commit with the JS fix.
Thanks for the review. :)

**Comment by bglusman** [comment]
Thanks man!  Hope you'll consider sticking around @fcapovilla !

**Pull request merged**
